src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -782,3 +782,14 @@ def validate_os_sku(namespace):
             CONST_OS_SKU_CBLMARINER,
             CONST_OS_SKU_MARINER,
         )
+
+
+def validate_azure_service_mesh_revision(namespace):
+    """Validates the user provided revision parameter for azure service mesh commands."""
+    if namespace.revision is None:
+        return
+    revision = namespace.revision
+    asm_revision_regex = re.compile(r'^asm-\d+-\d+$')
```

Comment:
does this regex account for if we were to have things like hotfixes/rc/beta etc. as a suffix? Basically a trailing wildcard for everything after asm-x-y*

Reply:
Will address this in a future pr, created a task for self - [Task 25420078](https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/25420078): Azure CLI changes for beta/hotfix release for ASM